packages/workers-editor-shared/vite.config.ts

```diff
@@ -0,0 +1,49 @@
+import { resolve } from "path";
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+import dts from "vite-plugin-dts";
+import pluginRewriteAll from "vite-plugin-rewrite-all";
+
+// See https://github.com/facebook/create-react-app/issues/11769
+// Turn react/jsx-runtime imports into react/jsx-runtime.js
+function addJsxRuntimeExtension() {
+	return {
+		name: "jsx-runtime-extension",
+		enforce: "pre",
+		resolveId(source) {
+			if (source === "react/jsx-runtime") {
+				return {
+					id: "react/jsx-runtime.js",
+					external: true,
+				};
+			}
+			return null;
+		},
+	};
+}
+
+// https://vitejs.dev/config/
+export default defineConfig({
+	plugins: [
+		pluginRewriteAll(),
+		react(),
+		dts({ rollupTypes: true }),
+		addJsxRuntimeExtension(),
```

Comment:
I'm seeing type errors on these, along these lines:

```
Type '{ name: string; configureServer(server: ViteDevServer): () => void; }' is not assignable to type 'PluginOption'.
```

Reply:
Looks like this was because `vite` wasn't a workspace dependency—added